client/src/components/dataset/sidebars/ChangeDataType.jsx

```diff
@@ -6,32 +6,23 @@ import SidebarControls from './SidebarControls';
 import DateFormatSelect from './DateFormatSelect';
 
 function DefaultValueInput({ defaultValue, onChange, newType }) {
+  const emptyValue = newType === 'number' ? 0 : '';
```

Comment:
This gets fed back to `DefaultValueInput` as `defaultvalue` -  now that it can be a number, we should probably change the propType of `defaultValue` from `PropTypes.string` to `PropTypes.oneOfType([PropTypes.string, PropTypes.number])` to suppress the console warning when it's 0